### PR TITLE
feat(tools-dgeni): add table of contents array to guides list

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
A table of contents is generated for all guides. The table of contents array is composed of the titles preceded by "## " in the associated markdown file.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export interface NavigationDocument {
   id: string;
   title: string;
   path: string;
+	tableOfContents: string[];
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts]: It wasn't supported before. I added support for it in the latest commit.

File: tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.ts [hunk 2]
```diff
@@ -12,8 +7,19 @@ export function guideFileReader() {
...
   };
+}
+
+const extractTitle = (doc: any) => {
+	const matchesArray = doc.content.match(/^\#(.*?)$/m);
+	if(!matchesArray){throw 'Guide doc ' + doc.id + ' is missing a title'};
+	return matchesArray ? matchesArray[1].trim() : '';
+}
+
+const extractTableOfContents = (doc: any) => {
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.ts]: done

**Comment by lderrickable** [comment]
@damienwebdev Changed to use `markdown-toc`

File: tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { extractTitle } from './guide-file.reader';
+
+describe('guideFileReader', () => {
+	
+	describe('extractTitle', () => {
```
Comment (damienwebdev) [inline tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts:5]: Can you add a test covering the result of toc?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts (hunk 3 above)
Comment (lderrickable) [reply tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts:5]: I don't think we should test the results of external packages.

Comment (damienwebdev) [reply tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts:5]: I think it's ok as long as we test only the API surface. If `toc` were to change, I would like to know if it violates our expectations.

Comment (lderrickable) [reply tools/dgeni/src/transforms/daffodil-guides-package/reader/guide-file.reader.spec.ts:56]: added

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**